### PR TITLE
Map carriers to channel

### PR DESCRIPTION
**Pull request by riteshshrv**

### CODE REVIEW

File: carrier.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    This model stores the carriers / shipping methods, each record
+    here can be mapped to a carrier in tryton which will then be
+    used for managing export of tracking info.
+    """
+    __name__ = 'channel.carrier'
+    _rec_name = 'title'
+
+    code = fields.Char("Code", readonly=True)
+    carrier = fields.Many2One('carrier', 'Carrier')
+    title = fields.Char('Title', readonly=True)
```
Comment (prakashpp) [inline carrier.py]: rename it to name, makes more sense

File: carrier.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    used for managing export of tracking info.
+    """
+    __name__ = 'channel.carrier'
+    _rec_name = 'name'
+
+    code = fields.Char("Code", readonly=True)
+    carrier = fields.Many2One('carrier', 'Carrier')
+    name = fields.Char('Name', readonly=True)
+    channel = fields.Many2One(
+        'sale.channel', 'Carrier Channel', readonly=True
```
Comment (tarunbhardwaj) [inline carrier.py]: Channel

File: carrier.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        Create shipping carriers by importing data from external channel.
+
+        Since external channels are implemented by downstream modules, it is
+        the responsibility of those channels to reuse this method or call super.
+
+        :param instances: Active record list of magento instances
+        """
+        pass
+
+    def get_shipping_carrier(self, code):
```
Comment (tarunbhardwaj) [inline carrier.py]: Move this method to Channel class

File: carrier.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+]
+
+
+class ChannelCarrier(ModelSQL, ModelView):
+    """
+    This model stores the carriers / shipping methods, each record
+    here can be mapped to a carrier in tryton which will then be
+    used for managing export of tracking info.
+    """
+    __name__ = 'channel.carrier'
```
Comment (tarunbhardwaj) [inline carrier.py]: sale.channel.carrier

File: carrier.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    This model stores the carriers / shipping methods, each record
+    here can be mapped to a carrier in tryton which will then be
+    used for managing export of tracking info.
+    """
+    __name__ = 'sale.channel.carrier'
+
+    name = fields.Char('Name', readonly=True)
+    code = fields.Char("Code", readonly=True)
+    carrier = fields.Many2One('carrier', 'Carrier')
```
Comment (tarunbhardwaj) [inline carrier.py]: just confirm if sale channel is dependent on carrier

File: channel.py [hunk 6]
```diff
@@ -114,6 +114,10 @@ class SaleChannel(ModelSQL, ModelView):
         "sale.channel.order_state", "channel", "Order States"
     )
 
+    channel_carriers = fields.One2Many(
```
Comment (tarunbhardwaj) [inline channel.py]: `shipping_carriers`

File: channel.py [hunk 7]
```diff
@@ -114,6 +114,10 @@ class SaleChannel(ModelSQL, ModelView):
...
+        'sale.channel.carrier', 'channel', 'Carriers / Shipping Methods'
```
Comment (tarunbhardwaj) [inline channel.py]: Shipping Carriers

File: channel.py [hunk 8]
```diff
@@ -239,6 +245,39 @@ def get_current_channel(cls):
...
+    def import_shipping_carriers(cls, channels):
+        """
+        Create shipping carriers by importing data from external channel.
+
+        Since external channels are implemented by downstream modules, it is
+        the responsibility of those channels to reuse this method or call super.
+
+        :param instances: Active record list of magento instances
+        """
+        raise NotImplementedError
```
Comment (tarunbhardwaj) [inline channel.py]: raise NotImplementedError() # instance

File: channel.py [hunk 9]
```diff
@@ -181,8 +185,10 @@ def __setup__(cls):
             'import_data_button': {},
             'export_data_button': {},
             'import_order_states_button': {},
+            'import_shipping_carriers': {},
         })
         cls._error_messages.update({
+            "no_carriers_found": "No carriers were found for code: %s",
```
Comment (tarunbhardwaj) [inline channel.py]: Shipping carrier is not configured for code: %s

**Pull request merged**